projects/canopy/src/lib/forms/input/docs/guide.stories.mdx

```diff
@@ -0,0 +1,160 @@
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <button lg-button lgSuffix size="sm">
+    Search
+  </button>
+</lg-input-field>
+```
+
+#### Icon Button suffix with add-on class
+
+There is an additional add-on button variant for adding a button to the input field which inherits the button spacing but no background or border colours.
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <button lg-button lgSuffix size="sm" [iconButton]="true" variant="add-on">
+    <lg-icon name="search" />
+  </button>
+</lg-input-field>
+```
+
+## Dos and Don'ts
+
+### Do
+
+![do](docs/text-input/do.png)
+
+1. **Do** use labels and supporting text to help provide context
+2. **Do** keep labels clear, concise and relevant
+3. **Do** use a textarea for multi-line content (Angular component to be built)
+4. **Do** use prefix, suffix or action variants to aid the user as to the input type or information required
+5. **Do** keep input widths relative to the information requested
+
+### Don't
+
+![dont](docs/text-input/dont.png)
+
+1. **Don't** truncate label text
+2. **Don't** use text inputs when a pre-defined response is required, use the select component instead
+3. **Don't** use text inputs for simple responses (such as yes/no), use checkboxes or radio buttons instead
+4. **Don't** use placeholder text instead of a label
+
+## Additional development information
+
+**We recommend using <a href="https://angular.io/guide/reactive-forms" target="_blank">Angular's reactive forms</a>.**
+
+### LgInputDirective inputs
+
+| Name | Description                                         | Type   | Default                      | Required |
+| ---- | --------------------------------------------------- | ------ | ---------------------------- | -------- |
+| ``id``   | HTML ID attribute, auto generated if not provided   | string | ``lg-input-${nextUniqueId++}`` | No       |
+| ``name`` | HTML Name attribute, auto generated if not provided | string | ``lg-input-${nextUniqueId++}`` | No       |
+
+### LgInputFieldComponent inputs
+
+| Name      | Description                                                  | Type    | Default                      | Required |
+| --------- | ------------------------------------------------------------ | ------- | ---------------------------- | -------- |
+| ``id``        | HTML ID attribute, auto generated if not provided. This will also propagate to the input 'id' and form 'for' attribute | string  | ``lg-input-${nextUniqueId++}`` | No       |
+| ``block``     | Property to make the input full width (for small screens only) | boolean | ``false``                        | No       |
+| ``showLabel`` | Show or visually hide the label | boolean | ``true`` | No       |
+
+------
+
+## Research on this component
+
+This component is based on best practice and has not had specific usability testing.
+
+## Related
+
+- [Select](./?path=/docs/components-form-select-guide--page)
+- Textarea
```

Comment:
does textarea not have a link?

Reply:
Hi, Walter. I couldn't find Textarea in the docs. Elena let me know that it is not developed yet. We'll need to update the link once textarea is done.

Reply:
is it worth chucking a note somewhere (into that ticket maybe)? wary that if we don't note it down anywhere, that it might get left behind

Reply:
Yes, that's sounds like a good idea, I'll put a note on the ticket.